src/word_debt_bot/game/core.py

```diff
@@ -55,3 +55,25 @@ def add_debt(self, player_id: str, amount: int):
         state = self._state
         state[player_id].word_debt += amount
         self._state = state
+
+    def create_leaderboard(self, sort_by: str, lb_len: int):
+        if sort_by not in ["debt", "cranes"]:
+            raise ValueError("ordering is done by 'debt' or 'cranes'")
+        if lb_len < 1:
+            raise ValueError("requested leaderboard length must be 1 or greater")
+        # Make a sort key and sort users
+        if sort_by == "debt":
+            key = lambda u: u[1].word_debt
+        elif sort_by == "cranes":
+            key = lambda u: u[1].cranes
+        users = sorted(self._state.items(), key=key, reverse=True)
```

Comment:
Nit: I think it'd be nicer to use `.values()` and to avoid indexing the tuple. For iteration you can also use [enumerate](https://docs.python.org/3/library/functions.html#enumerate) to avoid needing to manually keep track of the index when building the leaderboard string.

Reply:
Altered! That looks much nicer.